packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -2022,6 +2022,10 @@ export class Bucket extends BucketBase {
     function parseLifecycleRule(rule: LifecycleRule): CfnBucket.RuleProperty {
       const enabled = rule.enabled ?? true;
 
+      if ((rule.expirationDate || rule.expiration || rule.tagFilters) && rule.expiredObjectDeleteMarker) {
+        throw new Error('expiredObjectDeleteMarker cannot be true if expirationDate, expiration, or tagFilters are specified');
+      }
+
```

Comment:
This seems like we need a contract refactor and potential deprecation of props instead of throwing this error. Thoughts?

Reply:
Yes, there is a combinatorial explosion here caused by all these properties working together. I can't even think of a different contract that puts all these together in a coherent way. But I'm open to suggestions.